087-external-certificate-manager.md

```diff
@@ -0,0 +1,324 @@
+    generateSecretOwnerReference: <boolean>
+    renewalDays: <integer> # days before notAfter when we should start renewal
+    certificateExpirationPolicy: <renew-certificate|replace-key>
+    type: <strimzi.io|cert-manager.io> # (1)
+    certManagerIssuerRef: # (2)
+      name: <string>
+      kind: <Issuer|ClusterIssuer>
+      group: <string> # cert-manager.io by default
+    publicCert: # (3)
+      secretName: <string>
+      certificate: <string>
+```
+
+1. The`type` property will default to `strimzi.io` when not set and will use the existing behaviour, allowing backwards compatibility.
+   The option `cert-manager.io` will only be valid if `generateCertificateAuthority` is set to `false`.
+2. The property `certManagerIssuerRef` will only be used by Strimzi if `type` is set to `cert-manager.io`.
+  The `name`, `kind`, and `group` properties will be copied over into the `Certificate` custom resource Strimzi creates.
+3. The property `publicCert` will only be used by Strimzi if `type` is set to `cert-manager.io`.
+  The `secretName` and `certificate` properties will be used to locate the CA public certificate that must be trusted by Strimzi components in order to trust the end-entity certificates that cert-manager issues.
+
+### User steps
+
+To make use of this new option the user will have to:
+
+1. Install cert-manager.
+2. Create an `Issuer` or `ClusterIssuer` custom resource.
+3. Create a `Secret` containing the CA public cert for Strimzi to trust.
+   The `Secret` must contain a data entry with all the CAs bundled into one PEM file.
+   Users can optionally use [trust-manager](https://cert-manager.io/docs/trust/trust-manager/) to create this Secret, but they are responsible for installing trust-manager and creating the `Bundle` CR with a single `target` entry.
+4. Create a `Kafka` resource with `clusterCa.type` and/or `clientsCa.type` set to `cert-manager.io`, and `certManagerIssuerRef` and `publicCert` configured.
+
+### Internal handling of Cluster CA
+
+Strimzi uses two generation annotations to track how Cluster CA certificates are trusted and Cluster CA issued certificates are used:
+* `strimzi.io/cluster-ca-key-generation` on Kafka pods to indicate the generation of the CA private key that signed the CA public cert trusted by that pod
+* `strimzi.io/cluster-ca-cert-generation` on Secrets containing certificates to indicate the generation of the CA public cert at the time the certificate was issued
+
+The `strimzi.io/cluster-ca-cert-generation` annotation is also used on Kafka pods to indicate the generation of the CA public cert associated with the certificate it is currently presenting.
+
+The following sections go into more detail about how we will use these annotations when cert-manager is issuing certificates.
+At a high level:
+* Strimzi will use cert path validation using the `java.security` libraries to determine whether a new Cluster CA public cert was signed with a new private key.
```

Comment:
Should we make it clear that this can be done from the public keys only? (I assume it can?)